code/modules/reagents/chemistry/recipes.dm

```diff
@@ -339,56 +347,177 @@
+
+///////////END FIRE BASED EXPLOSIONS
+
+/*
+* Freezes in a circle around the holder location
+* Arguments:
+* * temp - the temperature to set the air to
+* * radius - the range of the effect
+* * freeze_duration - how long the icey spots remain for
+*/
+/datum/chemical_reaction/proc/freeze_radius(datum/reagents/holder, datum/equilibrium/equilibrium, temp, radius = 2, freeze_duration = 50)
+	for(var/any_turf in circlerangeturfs(center = get_turf(holder.my_atom), radius = radius))
+		if(!(istype(any_turf, /turf/open)))
+			continue
+		var/turf/open/open_turf = any_turf
+		open_turf.MakeSlippery(TURF_WET_PERMAFROST, 10, freeze_duration, freeze_duration)
+		open_turf.temperature = temp
+
+///Clears the beaker of the reagents only
+///if volume is not set, it will remove all of the reactant
+/datum/chemical_reaction/proc/clear_reactants(datum/reagents/holder, volume = 1000)
 	if(!holder)
 		return FALSE
-	for(var/datum/reagent/reagent as anything in holder.reagent_list)
-		if(!(reagent.type in required_reagents))
-			continue
-		if(!volume)
-			holder.remove_reagent(reagent.type, reagent.volume)
-		else
-			holder.remove_reagent(reagent.type, volume)
+	for(var/reagent in required_reagents)
+		holder.remove_reagent(reagent, volume)
 
-//Clears the beaker of the product only
+///Clears the beaker of the product only
 /datum/chemical_reaction/proc/clear_products(datum/reagents/holder, volume = null)
 	if(!holder)
 		return FALSE
-	for(var/datum/reagent/reagent as anything in holder.reagent_list)
-		if(!(reagent.type in results))
-			continue
-		if(!volume)
-			holder.remove_reagent(reagent.type, reagent.volume)
-		else
-			holder.remove_reagent(reagent.type, volume)
+	for(var/reagent in results)
+		holder.remove_reagent(reagent, volume)
+
 
-//Clears the beaker of ALL reagents inside
+///Clears the beaker of ALL reagents inside
 /datum/chemical_reaction/proc/clear_reagents(datum/reagents/holder, volume = null)
 	if(!holder)
 		return FALSE
+	if(!volume)
+		volume = holder.total_volume
 	holder.remove_all(volume)
+
+/*
+* "Attacks" all mobs within range with a specified reagent
+* Will be blocked if they're wearing proper protective equipment unless disabled
+* Arguments
+* * reagent - the reagent typepath that will be added
+* * vol - how much will be added
+* * range - the range that this will affect mobs for
+* * ignore_mask - if masks block the effect, making this true will affect someone regardless
+* * ignore_eyes - if glasses block the effect, making this true will affect someone regardless
+*/
+/datum/chemical_reaction/proc/explode_attack_chem(datum/reagents/holder, datum/equilibrium/equilibrium, reagent, vol, range = 2, ignore_mask = FALSE, ignore_eyes = FALSE)
+	if(istype(reagent, /datum/reagent))
+		var/datum/reagent/temp_reagent = reagent
+		reagent = temp_reagent.type
+	for(var/atom/movable/movey as anything in orange(range, get_turf(holder.my_atom)))
+		if(!(iscarbon(movey)))
+			continue
+		var/mob/living/carbon/target = movey
```

Comment:
```suggestion
	for(var/mob/living/carbon/target in orange(range, get_turf(holder.my_atom)))
```

The `as anything` pattern is for when we know type checks will not be needed. If they are needed, just use the normal syntax.